src/main/java/gov/cabinetoffice/gap/service/ZipService.java

```diff
@@ -109,19 +99,23 @@ private static void downloadFile(final String fileName) {
         }
     }
 
-    public static String parseFileName(final String fileName, int suffix) {
-        String[] subString = fileName.split("/");
-        logger.info("subString " + subString);
-        String[] parts = subString[subString.length - 1].split("\\.");
-        logger.info("parts " + parts);
-
-        String name = parts[0];
-        logger.info("name " + name);
-        String extension = parts.length > 1 ? "." + parts[1] : "";
-        logger.info("extension " + extension);
-
-        logger.info("name concat " + name.concat("_" + suffix + extension));
-        return name.concat("_" + suffix + extension);
+    public static String parseFileName(final String objectKey, int suffix, final String applicationId,
+                                       final String submissionId) {
+        //an object key is formed by applicationId/submissionId/s3bucketRandomFolderName/filename
```

Comment:
could we drop the logic from line 104 - 108 into its own method? Perhaps name it something like `getFileNameFromS3ObjectKey`